src/java/org/apache/cassandra/index/sai/disk/vector/VectorMemtableIndex.java

```diff
@@ -156,9 +158,15 @@ public void update(DecoratedKey key, Clustering clustering, ByteBuffer oldValue,
 
             // make the changes in this order so we don't have a window where the row is not in the index at all
             if (newRemaining > 0)
+            {
                 graph.add(newValue, primaryKey);
+                overwriteCount.increment();
+            }
             if (oldRemaining > 0)
+            {
                 graph.remove(oldValue, primaryKey);
+                removedCount.increment();
```

Comment:
I was focused on which methods we were calling on the `graph`. But I see that the counts produced won't be as helpful since we're double counting in some cases and not others. I think you're right that we should move it to the next conditional block.